pubnub-core/pubnub-core-api/src/main/kotlin/com/pubnub/api/v2/BasePNConfiguration.kt

```diff
@@ -581,5 +581,26 @@ interface BasePNConfigurationOverride {
          * If set, all communications to and from PubNub will be encrypted.
          */
         val cryptoModule: CryptoModule?
+
+        /**
+         * How long before the client gives up trying to connect with the server.
+         *
+         * The value is in seconds.
+         *
+         * Defaults to 5.
+         */
+        val connectTimeout: Int
+
+        /**
+         * For non subscribe operations (publish, herenow, etc),
+         * This property relates to a read timeout that is applied from the moment the connection between a client
+         * and the server has been successfully established. It defines a maximum time of inactivity between two
```

Comment:
This is not critical for now.

Currently we have property name "nonSubscribeRequestTimeout" that is a bit misleading because name suggests that this property relates to requestTimeout that can be understood like the maximum time a client will wait for a response from a server after it sends a request. This includes:
-resolving the hostname
-establishing a TCP connection
-TLS handsake
-sending the request data
-receiving the entire response data

In fact in our SDK "nonSubscribeRequestTimeout" controls read timeout that is applied from the moment the connection between a client and a server has been successfully established. It defines a maximum time of inactivity between two data packets when waiting for the servers's response.

It would be good to deprecate "nonSubscribeRequestTimeout" and introduce something like "nonSubscribeReadTimeout" instead.

This is not critical for now. Just want to share my observation.

Reply:
done